### ISSUE TITLE
tf.math.acos should accept only float/double inputs

### ISSUE THREAD

**Issue by lugalUrim** [tensorflow/tensorflow#48467]
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04.3
Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
TensorFlow installed from (source or binary): binary
TensorFlow version: 2.4.1
Python version: 3.7
Installed using virtualenv? pip? conda?: pip
Bazel version (if compiling from source): N/A
GCC/Compiler version (if compiling from source): N/A
CUDA/cuDNN version: N/A
GPU model and memory: N/A

## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/math/acos

### Parameters defined

The input tensor `x` can not be bfloat16, half, uint8, int8, int16, int32, int64, complex64, complex128, string. In fact, only float32 and float64 is supported now.

### Usage example

```
import tensorflow as tf
a = tf.constant([1.0], dtype = tf.half)
tf.math.acos(a)
```

Output:
NotFoundError: Could not find device for node: {{node Acos}} = Acos[T=DT_INT8]
All kernels registered for op Acos:
  device='CPU'; T in [DT_DOUBLE]
  device='CPU'; T in [DT_FLOAT]
  device='GPU'; T in [DT_DOUBLE]
  device='GPU'; T in [DT_FLOAT]
 [Op:Acos]

**Comment by tilakrayal** [comment]
@lugalUrim ,


In order to expedite the trouble-shooting process, could you please provide the following information
OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
TensorFlow installed from (source or binary):
TensorFlow version:
Python version:
Installed using virtualenv? pip? conda?:
Bazel version (if compiling from source):
GCC/Compiler version (if compiling from source):
CUDA/cuDNN version:
GPU model and memory:
 
and the exact sequence of commands / steps that you executed before running into the problem

Thanks!

**Comment by lugalUrim** [comment]
Updated. Thanks. @tilakrayal

**Comment by tilakrayal** [comment]
I am able to replicate the issue on tf 2.4 and nightly but the issue is not reproducible on tf 2.3, please find the [gist](https://colab.research.google.com/gist/tilakrayal/b4a87b4a6a7d459e9cc39cb3ecc4756c/48467.ipynb) here.

**Comment by ymodak** [comment]
This is fixed with TF 2.6 please refer colab [gist](https://colab.research.google.com/gist/ymodak/7336b5b143ee04f7c8e88d50eb51e606/untitled3.ipynb). Thanks!
